app/controllers/sign_up/select_email_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module SignUp
+  class SelectEmailController < ApplicationController
+    before_action :confirm_two_factor_authenticated
+
+    def show
+      @sp_name = sp_name
+      @user_emails = user_emails
+      @last_sign_in_email_address = last_email
+      @select_email_form = build_select_email_form
+    end
+
+    def create
+      @select_email_form = build_select_email_form
+
+      result = @select_email_form.submit(form_params)
+      if result.success?
+        session[:selected_email_id] = form_params[:selected_email_id]
+        redirect_to sign_up_completed_path
+      else
+        flash[:error] = t('anonymous_mailer.password_reset_missing_user.subject')
```

Comment:
Same comment as before about not repurposing text from mailers. Usually we should be able to pull the specific error message from the form result itself, using `result.first_error_message`.

(This should probably be documented better in our [forms documentation](https://github.com/18F/identity-idp/blob/main/docs/backend.md#forms-formresponse-analytics-and-controllers), but the general idea being that the message we're using in forms validation should be the one we expect to show up in the user interface)